File: Pod/Classes/ABEIssue.m
//
//  ABEIssue.m
//  IssueReporter
//
//  Created by Nikolai Johan Heum on 04.05.15.
//  Copyright (c) 2015 Abello. All rights reserved.
//

#import "ABEIssue.h"

#import "ABEReporter.h"

@import Foundation;

@interface ABEIssue ()
@property (nonatomic) NSMutableArray *imageURLs;
@end

@implementation ABEIssue

- (NSDictionary *)toDictionary {
    return @{NSStringFromSelector(@selector(title)) : self.title,
             NSStringFromSelector(@selector(body)) : self.body};
}

- (void)attachImageAtURL:(NSString *)url {
    [self.imageURLs addObject:url];
}

- (NSString *)body {
    NSString *base = [NSString stringWithFormat:@"%@ \n\n %@ \n", self.issueDescription, [[ABEReporter reporter] extraInfoForIssue]];
    
    for (NSString *url in self.imageURLs) {
        base = [base stringByAppendingString:[NSString stringWithFormat:@"![image](%@)\n", url]];
    }
    
    return base;
}

- (NSMutableArray *)imageURLs {
    if (!_imageURLs) {
        _imageURLs = [NSMutableArray new];
    }
    
    return _imageURLs;
}

@end

File: Pod/Classes/ABEIssue.swift
//
//  ABEIssue.swift
//  Pods
//
//  Created by Hakon Hanesand on 10/6/16.
//
//

struct ABEIssue {
   
    var title: String?
    var issueDescription: String?
    
    var imageURLS : [String] = []
    
    mutating func attachImage(withURL urlString: String) {
        imageURLS.append(urlString)
    }
    
    var textRepresentation : String? {
        // TODO
        let extraDebuggingInformation = ""
        
        let base = "\(issueDescription) \n\n \(extraDebuggingInformation)"
        let images = imageURLS.map { "![image](\($0))\n" }.reduce("") { $0 + "\n" + $1 }
        
        print(images)
        
        return base + "\n" + images
    }
    
    var dictionaryRepresentation : [String : String]? {
        guard let title = title, let body = textRepresentation else { return nil }
        return ["title" : title, "body" : body]
    }
}
